build.rs

```diff
@@ -1,7 +1,8 @@
 use std::env;
 
 fn main() {
-    let force_scalar_math = env::var("CARGO_FEATURE_SCALAR_MATH").is_ok();
+    let force_scalar_math = env::var("CARGO_FEATURE_SCALAR_MATH").is_ok()
+        || env::var("CARGO_CFG_TARGET_ARCH").unwrap() == "spirv";
```

Comment:
I would have though your target would get scalar math anyway as it lacks SSE2, or is that not the case? The other thing this does is turns off align16 on a bunch of types. Is that what you are wanting to achieve here?

Reply:
the SSE2 support did get enabled when I didn't have this, maybe it is still somehow in the `CARGO_CFG_TARGET_FEATURE`, haven't debugged it more carefully so just did a quick solution here.

Was intending of not having any alignment and using "scalar" types

Reply:
Tried debugging this to figure out what's going on here, and could repro it _once_, but then once I `cargo clean`'d and rebuilt with a bunch of debugging `println!` whatnots, I... could no longer repro it, no matter what I tried.

I'm guessing it may be a build cache bug, possibly due to lack of [cargo:rerun-if-env-changed=NAME](https://doc.rust-lang.org/cargo/reference/build-scripts.html#rerun-if-env-changed) on these variables, but that's a wild guess. In any case, I'm going to remove this as it seems to work fine without it.

Reply:
Right, managed to repro it now, the actual problem is that we're building `rust-gpu` crates in a nested cargo process. Cargo, helpfully, if there are no target features, does not reset-to-empty/unset `CARGO_CFG_TARGET_FEATURE`. So that means the outer cargo process's `CARGO_CFG_TARGET_FEATURE` "leaks" into the inner process, causing `CARGO_CFG_TARGET_ARCH` to say `spirv` but `CARGO_CFG_TARGET_FEATURE` has x86_64 features. Ugh.

This is our own problem to fix, though, not glam's, so not having this bit of code in glam is the right way to go.